### PR TITLE
Sampling on Manifolds

### PR DESCRIPTION
**Pull request by nguigs**
- Fix the random uniform method of SO(n)
- Generalize the von Mises-Fisher distribution to any dimension and any mean point

### CODE REVIEW

File: geomstats/geometry/special_orthogonal.py [hunk 1]
```diff
@@ -335,6 +336,56 @@ def vector_from_skew_matrix(self, skew_mat):
         """
         return SkewSymmetricMatrices(self.n).basis_representation(skew_mat)
 
+    def to_tangent(self, vector, base_point=None):
+        return self.regularize_tangent_vec(vector, base_point)
+
+    @geomstats.vectorization.decorator(
+        ['else', 'vector', 'else', 'output_point'])
+    def regularize_tangent_vec_at_identity(
```
Comment (ninamiolane) [inline geomstats/geometry/special_orthogonal.py:344]: Hm I know this is not your code, but I don't remember the purpose of `regularize`, `regularize_tangent_vec_at_identity` and `regularize_tangent_vec`: is it expected that they all call `regularize`?

File: geomstats/geometry/special_orthogonal.py [hunk 2]
```diff
@@ -335,6 +336,56 @@ def vector_from_skew_matrix(self, skew_mat):
...
+            Tangent vector at base point.
+        base_point : array-like, shape=[..., 1]
+            Point on the manifold.
+
+        Returns
+        -------
+        regularized_tangent_vec : array-like, shape=[..., 1]
+            Regularized tangent vector.
+        """
+        return self.regularize_tangent_vec_at_identity(tangent_vec)
```
Comment (ninamiolane) [inline geomstats/geometry/special_orthogonal.py:387]: Could this be `self.regularize(tangent_vec)` to save one call?

File: tests/test_hypersphere.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 from geomstats.geometry.hypersphere import Hypersphere
 from geomstats.geometry.matrices import Matrices
 
-MEAN_ESTIMATION_TOL = 1e-6
+MEAN_ESTIMATION_TOL = 1e-3
```
Comment (ninamiolane) [inline tests/test_hypersphere.py:10]: Why this change in tolerance?

**Review by ninamiolane — APPROVED** [review]
Awesome, thanks! 🎉  Just one question on some code that isn't yours (about `regularize`), and another small comment but it's already good to go and I'll let you merge!

File: geomstats/geometry/special_orthogonal.py (hunk 1 above)
Comment (nguigs) [reply geomstats/geometry/special_orthogonal.py:344]: In vector representation, points are in fact tangent vectors so there's no surprise all these methods are the same.

File: geomstats/geometry/special_orthogonal.py (hunk 2 above)
Comment (nguigs) [reply geomstats/geometry/special_orthogonal.py:387]: Yes

File: tests/test_hypersphere.py (hunk 3 above)
Comment (nguigs) [reply tests/test_hypersphere.py:10]: Because it doesn't work that well when changing the mean... But I guess this is expected from properties of an estimator, given that according to the central limit theorem, the sample mean has variance in sqrt(n), and n is 10^6 here.

File: geomstats/geometry/special_orthogonal.py (hunk 2 above)
Comment (nguigs) [reply geomstats/geometry/special_orthogonal.py]: DeepSource goes crazy when I try, because of the unused arguments

Comment (nguigs) [reply geomstats/geometry/special_orthogonal.py:384]: In fact Deepsource didn't raise an error for the unused arguments before because they were used by the decorator... What should we do?

**Comment by ninamiolane** [comment]
@nguigs sorry that you had to go through this DeepSource problem... Let's forget about DeepSource, I think we should remove it and autoformat the codebase with `black`, see issue #918
